### ISSUE TITLE
Add command: prefect server stop

### ISSUE THREAD

**Issue by joelluijmes** [PrefectHQ/prefect#3660]
## Current behavior

We can locally start server with `prefect server start`, docs: https://docs.prefect.io/api/latest/cli/server.html
But unless I'm mistaken, I miss the reverse, the stop command.

## Proposed behavior

Add stop command which stops prefect.

**Comment by rmax** [comment]
I see that the CLI uses [tempfile.gettempdir()](https://github.com/PrefectHQ/prefect/blob/0e9fa02c7acac586610c9894bec82e90fdf57c8b/src/prefect/cli/server.py#L266) as temporary location of the modified `docker-compose.yml`.

That location is consistent between runs (at least in OSX) so it's possible to reference the file for a `docker-compose down` command. Although, it might be good to use a subidrectory rather than the root temp dir to avoid potential side effects.

By using the `--file` option I can stop the containers. Notice the containers are prefixed by 't_' which comes from the 'T' directory name. This could be more friendly by using `--project prefect` or similar in both `up` and `down` commands to get the container names more descriptive.

```
❯ docker ps --format '{{.Names}}\t{{.Image}}'
t_ui_1  prefecthq/ui:core-0.13.14
t_towel_1       prefecthq/server:core-0.13.14
t_apollo_1      prefecthq/apollo:core-0.13.14
t_graphql_1     prefecthq/server:core-0.13.14
t_hasura_1      hasura/graphql-engine:v1.3.0
t_postgres_1    postgres:11
```

Here is how I stop the containers:

```
❯ python -c 'import tempfile; print(tempfile.gettempdir())'
/var/folders/nv/7pj04vn56wlf1470pj2_3pth0000gn/T                                    

❯ ls /var/folders/nv/7pj04vn56wlf1470pj2_3pth0000gn/T | head
9B8F6828-49B5-46D9-B42F-085082C0528A
AppTranslocation
AudioComponentRegistrar
DMD01241D7D-D786-4D42-89D1-519D259E13F0
DMD01AD8FA0-ADE9-4D14-8801-3298810A7185
DMD02472AF6-CC86-416A-8EF5-CAA7206C2AF4
DMD02C59915-9D0C-45AA-BDC8-1C058BC9B822
DMD0418B6C5-C841-46C3-8C33-107D40CCFB53
DMD048794A7-3598-462C-9B98-F74B7E44D3CA                                             

❯ head /var/folders/nv/7pj04vn56wlf1470pj2_3pth0000gn/T/docker-compose.yml
networks:
  prefect-server:
    name: prefect-server
services:
  apollo:
    command: bash -c "./post-start.sh && npm run serve"
    depends_on:
    - graphql
    environment:
      GRAPHQL_SERVICE_HOST: http://graphql                                          

❯ docker-compose --file /var/folders/nv/7pj04vn56wlf1470pj2_3pth0000gn/T/docker-compose.yml down
WARNING: The PREFECT_SERVER_DB_CMD variable is not set. Defaulting to a blank string.
WARNING: The DB_CONNECTION_URL variable is not set. Defaulting to a blank string.
WARNING: The POSTGRES_DB variable is not set. Defaulting to a blank string.
WARNING: The POSTGRES_PASSWORD variable is not set. Defaulting to a blank string.
WARNING: The POSTGRES_USER variable is not set. Defaulting to a blank string.
Stopping t_ui_1       ... done
Stopping t_towel_1    ... done
Stopping t_apollo_1   ... done
Stopping t_graphql_1  ... done
Stopping t_hasura_1   ... done
Stopping t_postgres_1 ... done
Removing t_ui_1       ... done
Removing t_towel_1    ... done
Removing t_apollo_1   ... done
Removing t_graphql_1  ... done
Removing t_hasura_1   ... done
Removing t_postgres_1 ... done
Removing network prefect-server       
```

However, this approach doesn't stop the `prefect server start` process which keeps running until I hit CTRL-C:
```
graphql_1   | INFO:     172.22.0.5:56310 - "POST /graphql/ HTTP/1.1" 200 OK
graphql_1   | INFO:     172.22.0.5:56314 - "POST /graphql/ HTTP/1.1" 200 OK
t_ui_1 exited with code 0
t_apollo_1 exited with code 0
t_towel_1 exited with code 137
graphql_1   | INFO:     Shutting down
graphql_1   | INFO:     Waiting for application shutdown.
graphql_1   | INFO:     Application shutdown complete.
graphql_1   | INFO:     Finished server process [1]
t_graphql_1 exited with code 0
t_hasura_1 exited with code 0
postgres_1  | 2020-11-13 15:27:22.673 UTC [1] LOG:  received fast shutdown request
postgres_1  | 2020-11-13 15:27:22.674 UTC [1] LOG:  aborting any active transactions
postgres_1  | 2020-11-13 15:27:22.677 UTC [1] LOG:  background worker "logical replication launcher" (PID 71) exited with exit code 1
postgres_1  | 2020-11-13 15:27:22.678 UTC [66] LOG:  shutting down
postgres_1  | 2020-11-13 15:27:22.738 UTC [1] LOG:  database system is shut down
t_postgres_1 exited with code 0
^CException caught; killing services (press ctrl-C to force)
Removing network prefect-server
WARNING: Network prefect-server not found.

Aborted!                                      
```

Another alternative would be saving the PID of the `prefect server` process to a file and then on `prefect server stop` use it to send a `SIGINT` to the process which would make docker to stop the containers.

For example:
```
❯ nohup prefect server start 2>&1 >prefect-server.log &

❯ echo $! > prefect-server.pid                                                     

❯ cat prefect-server.pid
91912                                                                              

❯ tail prefect-server.log
apollo_1    | Server ready at http://0.0.0.0:4200 🚀 (version: master)
apollo_1    | Sending telemetry to Prefect Technologies, Inc.: {"source":"prefect_server","type":"startup","payload":{"id":"4a6258b9-112f-4b77-8e76-675084261537","prefect_server_version":"master","api_version":"0.2.0"}}
graphql_1   | INFO:     172.25.0.6:53608 - "POST /graphql/ HTTP/1.1" 200 OK
graphql_1   | INFO:     172.25.0.6:53620 - "POST /graphql/ HTTP/1.1" 200 OK
graphql_1   | INFO:     172.25.0.6:53650 - "POST /graphql/ HTTP/1.1" 200 OK
graphql_1   | INFO:     172.25.0.6:53652 - "POST /graphql/ HTTP/1.1" 200 OK
graphql_1   | INFO:     172.25.0.6:53654 - "POST /graphql/ HTTP/1.1" 200 OK
graphql_1   | INFO:     172.25.0.6:53700 - "POST /graphql/ HTTP/1.1" 200 OK
graphql_1   | INFO:     172.25.0.6:53704 - "POST /graphql/ HTTP/1.1" 200 OK
graphql_1   | INFO:     172.25.0.6:53708 - "POST /graphql/ HTTP/1.1" 200 OK        

❯ kill -SIGINT $(cat prefect-server.pid)                                           

❯ tail prefect-server.log
Exception caught; killing services (press ctrl-C to force)
Stopping t_ui_1       ... done
Stopping t_apollo_1   ...
Stopping t_apollo_1   ... done
Stopping t_graphql_1  ...
Stopping t_hasura_1   ...
Stopping t_postgres_1 ...
t_ui_1 exited with code 0
t_apollo_1 exited with code 0                                                      
Removing t_apollo_1   ... done
Removing t_towel_1    ... done
Removing t_graphql_1  ... done
Removing t_hasura_1   ... done
Removing t_postgres_1 ... done
Removing network prefect-server

Aborted!             
```

**Comment by madkinsz** [comment]
I've seen the "persist PID to a file" pattern several times before and it makes sense to me, just my 2cents.

**Comment by joelluijmes** [comment]
Ah that explains, tried to look into it to see if it was trivial, but didn’t find any meaningful docker-compose references. Isn’t the most easy solution, not to use a temporary directory but write the docker-comopse to a fixed location? E.g. somewhere in `~/prefect`

**Comment by limx0** [comment]
I think I've used something like `docker network inspect prefect-server | jq -r '.[0].Containers | map(.Name) | join("\n")' | xargs docker rm -f` with some success.

**Comment by jcrist** [comment]
Fixed by #3899.
